packages/evolution-backend/src/api/survey.user.routes.ts

```diff
@@ -119,15 +128,54 @@ export default (authorizationMiddleware, loggingMiddleware: InterviewLoggingMidd
                         valuesByPath,
                         unsetPaths,
                         serverValidations: serverConfig.serverValidations,
-                        fieldsToUpdate: ['responses', 'validations']
+                        fieldsToUpdate: ['responses', 'validations'],
+                        deferredUpdateCallback: async (newValuesByPath) => {
+                            // Reload the session as this can be called much later
+                            session.reload((err) => {
+                                if (err !== undefined) {
+                                    console.error(`Error reloading session: ${err}`);
+                                    return;
+                                }
+                                const reloadedSession = req.session;
+                                // Save the updated paths in the current user's
+                                // session data for asynchronous update to the
+                                // client, but make sure the interview id is the
+                                // same as the one is was saved for
+                                const currentSessionValues = reloadedSession.clientValues || {
+                                    interviewId: interview.uuid,
+                                    updatedPaths: []
+                                };
+                                const updatedPaths =
+                                    currentSessionValues.interviewId === interview.uuid
```

Comment:
Can you put more comments on this part?

Reply:
I added some more meat to the comment of line 140, which explains the function's behavior